src/System.Management.Automation/engine/interpreter/InstructionFactory.cs

```diff
@@ -124,5 +126,19 @@ protected internal override Instruction NewArrayInit(int elementCount)
             }
             return new NewArrayInitInstruction<T>(elementCount);
         }
+
+        protected internal override Instruction NewListInit(int elementCount)
```

Comment:
I don't think we should add this to the interpreter - this way it stays as close as possible to original and/or the one used for `dynamic` when now JIT is available.

It is possible to add new instructions without touching the interpreter it that's how you want to do it - see `PowerShellLoopExpression` for an example.

Reply:
The `NewListInitInstruction` is a minor optimization which can be removed.
But I need to implement `LightCompiler.CompileListInitExpression` because it's currently not implemented. Are you suggesting to not use `ListInitExpression` in our compiler generated code?

Reply:
I see the method is now implemented: http://source.dot.net/#System.Linq.Expressions/System/Linq/Expressions/Interpreter/LightCompiler.cs,a988c37e1da7a799

I wonder if there is a supported way of invoking the interpreter now? We'd have to make sure they have any of our fixes first, but it'd be nice to not ship two copies of basically the same thing.

Reply:
Great, I can use their implementation then.